waiter/src/waiter/util/timer_context.clj

```diff
@@ -0,0 +1,23 @@
+;;
+;;       Copyright (c) 2017 Two Sigma Investments, LP.
+;;       All Rights Reserved
+;;
+;;       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF
+;;       Two Sigma Investments, LP.
+;;
+;;       The copyright notice above does not evidence any
+;;       actual or intended publication of such source code.
+;;
+(ns waiter.util.timer-context
+  (:require [clj-time.core :as t])
+  (:import (com.codahale.metrics Timer)
+           (java.util.concurrent TimeUnit)))
+
+(defrecord TimerContext [timer start-time])
+
+(defn report-duration
```

Comment:
That's where I originally wanted to put it, but then we end up with a cyclic dependence between `waiter.metrics` and `waiter.util.utils` since `TimerContext` would be defined in the former and used in the latter (but `waiter.metrics` already depends on `waiter.util.utils`).

Reply:
We end up avoiding that dependence if the query channel modifies the timer objects when it responds to queries :)

Reply:
I got rid of the `TimerContext` record in favor of storing the start-times on their own. The `report-duration` function is now in the `waiter.metrics` namespace as you suggested.